src/server/index.tsx

```diff
@@ -48,9 +49,73 @@ export const createServer = ({
   rootPath,
   _injectAuthenticationToken = injectAuthenticationToken,
 }: CreateServerParams) => {
+  // This CSP is a tight default. This CSP should be on all requests
+  // that aren't serving a document. This CSP should be set with statics.
+  const baseCSP = {
+    defaultSrc: ["'none'"],
+    baseUri: ["'none'"],
+    formAction: ["'none'"],
+    objectSrc: ["'none'"],
+    reportUri: '/__cspreport__',
+  };
+
+  // This config sets the non-static CSP for deployed instances.
+  // TODO: A separate CDN should be configured for statics. When that happens
+  // the 'self' should be removed and replaced with the CDN host + path for statics.
+  const prodCSP = {
+    defaultSrc: ["'none'"],
+    childSrc: ["'none'"],
+    connectSrc: [
+      // Relax the connect-src if using the proxy otherwise
+      // Use the env var or 'none' if the API host isn't set.
+      env.REACT_APP_USE_INSECURE_PROXY === 'true'
+        ? "'self'"
+        : env.REACT_APP_API_HOST || "'none'",
+    ],
+    baseUri: ["'self'"],
+    fontSrc: ["'none'"],
+    formAction: ["'none'"],
+    frameAncestors: ["'none'"],
+    frameSrc: ["'none'"],
+    imgSrc: ["'self'"],
+    manifestSrc: ["'none'"],
+    mediaSrc: ["'none'"],
+    objectSrc: ["'none'"],
+    scriptSrc: ["'self'"],
+    styleSrc: ["'self'"],
+    workerSrc: ["'none'"],
+    reportUri: '/__cspreport__',
+  };
+
+  // This CSP is the one above with overrides for local development.
+  const localCSP = {
```

Comment:
Could this definition move to the exact place it's needed? which I think is here: https://github.com/mozilla/addons-code-manager/blob/9ca65f81b8daadf29dd6f643ac8418ac32ac8413/src/server/index.tsx#L203

Since it's only used in that place, I think it would help keep things concise.